sql/core/src/main/scala/org/apache/spark/sql/execution/basicOperators.scala

```diff
@@ -110,23 +114,28 @@ case class Filter(condition: Expression, child: SparkPlan)
   override def doConsume(ctx: CodegenContext, input: Seq[ExprCode], row: String): String = {
     val numOutput = metricTerm(ctx, "numOutputRows")
 
-    // filter out the nulls
-    val filterOutNull = notNullAttributes.map { a =>
-      val idx = child.output.indexOf(a)
-      s"if (${input(idx).isNull}) continue;"
-    }.mkString("\n")
+    val conjuncts = splitConjunctivePredicates(condition)
 
-    ctx.currentVars = input
-    val predicates = otherPreds.map { e =>
-      val bound = ExpressionCanonicalizer.execute(
-        BindReferences.bindReference(e, output))
-      val ev = bound.gen(ctx)
+    // Generate the code to evaluate each filter.
+    val generated = conjuncts.map { c =>
```

Comment:
It's better to use `notNullAttributes  ++ otherPreds` to make sure that `IsNotNull` always come first.